packages/boot/test/bootstrapTests/net-ibc-upgrade.test.ts

```diff
@@ -94,109 +100,72 @@ test.serial('upgrade at many points in network API flow', async t => {
-      return [label, { ...opts, serverAddress }];
+      return { ...opts, serverAddress };
     },
-    requestConnection: async ([label, opts]) => {
+    requestConnection: async opts => {
       await EV.sendOnly(opts.client).connect(opts.serverAddress);
-      return [label, opts];
+      return opts;
     },
-    acceptConnection: async ([label, opts]) => {
+    acceptConnection: async opts => {
       await EV.sendOnly(opts.server).dequeue('onAccept');
-      return [label, opts];
+      return opts;
     },
-    openConnection: async ([label, opts]) => {
+    openConnection: async opts => {
       await EV.sendOnly(opts.server).dequeue('onOpen');
-      return [label, opts];
+      return opts;
     },
-    sendPacket: async ([label, opts]) => {
+    sendPacket: async (opts, label) => {
       await EV.sendOnly(opts.client).send(label);
-      return [label, opts];
+      return opts;
     },
-    respond: async ([label, opts]) => {
+    respond: async opts => {
       await EV.sendOnly(opts.server).dequeue('onReceive');
-      return [label, opts];
+      return opts;
     },
-    checkAck: async ([label, opts]) => {
+    checkAck: async (opts, label) => {
       const ack = await EV(opts.client).getAck();
       t.is(ack, `got ${label}`, `${label} expected echo`);
-      return [label, { ...opts, ack }];
+      return { ...opts, ack };
     },
-    closeConnection: async ([_label, opts]) => {
+    closeConnection: async opts => {
       await EV(opts.client).close();
     },
-  });
-
-  const doSteps = async (label, steps, input: unknown = undefined) => {
-    await null;
-    let result = input;
-    for (const [stepName, fn] of steps) {
-      await t.notThrowsAsync(async () => {
-        result = await fn(result);
-      }, `${label} ${stepName} must complete successfully`);
-    }
-    return result;
-  };
-
-  // Sanity check
-  await doSteps('pre-upgrade', flow, 'pre-upgrade');
-
-  // For each step, run to just before that point and pause for
-  // continuation after upgrade.
-  const pausedFlows = [] as Array<{
-    result: any;
-    remainingSteps: [string, (lastResult: any) => unknown][];
-  }>;
-  for (let i = 0; i < flow.length; i += 1) {
-    const [beforeStepName] = flow[i];
-    const result = await doSteps(
-      `pre-${beforeStepName}`,
-      flow.slice(0, i),
-      `pause-before-${beforeStepName}`,
-    );
-    pausedFlows.push({ result, remainingSteps: flow.slice(i) });
-  }
+  } satisfies Record<string, TestStep[1]>);
 
-  await upgradeVats(t, EV, ['ibc', 'network']);
-
-  // Verify a complete run post-upgrade.
-  await doSteps('post-upgrade', flow, 'post-upgrade');
-
-  // Verify completion of each paused flow.
-  for (const { result, remainingSteps } of pausedFlows) {
-    const [beforeStepName] = remainingSteps[0];
-    await doSteps(`resumed-${beforeStepName}`, remainingSteps, result);
-  }
+  await testInterruptedSteps(t, allSteps, async () => {
```

Comment:
I see the value in `testInterruptedSteps` but I'm I'm less sure about it going in internal when there's only one test that uses it. Consider for now keeping it in `@aglocal/boot`.

Reply:
I was torn about this (had it just in `@aglocal/boot` before), but rationalized putting it in internal because there's nothing about it that's specific to boot.